src/mempool/README.md

```diff
@@ -0,0 +1,70 @@
+# Mempool
+
+In Cardano, for blocks to have useful data, they have to contain transactions,
+which are codifications of operations on the Ledger state that only some
+authorized actors can enact. Notice that such transactions are what makes up the
+contents of the block. In one way, one could see the ledger as being able to
+validate transactions, and by implication full blocks as those are mainly
+collections of transactions.
+
+The Mempool is the abstract component of the node that stores transactions which
+are valid on top of the latest known Ledger State (the one at the tip of the
+chain selected by [Chain Selection](../consensus/chainsel.md) in the Consensus
+layer). The validity of transactions is defined by the Ledger layer. Notice a
+change in the selected chain should trigger a prompt revalidation of the pending
+transactions to discard those that became invalid on the new selection.  Such
+collection of valid transactions will be requested by the Consensus layer to
+[forge new blocks](../consensus/forging.md).
+
+Being valid implies in particular that the transaction is pending, i.e. it has
+not been included in any prior block ever, as in Cardano every valid transaction
+must consume at least one UTxO.
+
+> [!NOTE]
+>
+> Notice that a transaction might be considered valid in one peer, but invalid
+> in another peer which has selected a different chain. Even in the case both
+> peers have the same selection, the existence in the mempool (or lack thereof)
+> of some other specific transaction that creates (or consumes) the inputs for
+> this one could lead to different veredicts on each peer.
+
+While only nodes that forge new blocks will use these transactions to create
+blocks, non-block-producing nodes are still expected to diffuse transactions to
+their peers. **All participating nodes** must be able to receive, validate, and
+distribute transactions.
+
+> [!TIP]
+>
+> When a node is shut down, the implementation can choose what to do with the
+> pending transactions. The reference Haskell implementation discards them.
+> However, it is conceivable for a node to store on disk the pending
+> transactions from its mempool.
+
+The [`TxSubmission2`](./txsubmission2.md) miniprotocol is the
+one used to diffuse the pending transactions through the network. Any mempool
+implementation has some requirements to allow diffusing transactions via
```

Comment:
I wonder if this sentence could be improved. I had to read it several times and I'm still unsure if I got it right :sweat_smile: Is it saying something like: a mempool implementation has to satisfy these requirements ... this allows the mempool impl to diffuse via tx2 and support the block forging functionality of consensus?

Reply:
Reworded